src/tools/r2rdump/R2RMethod.cs

```diff
@@ -203,84 +202,41 @@ public R2RMethod(byte[] image, MetadataReader mdReader, uint rid)
                 ArgTypes[i] = new SignatureType(ref signatureReader, mdReader, genericParams);
             }
 
-            Token = _mdtMethodDef | rid;
+            if (IsGeneric && instanceArgs != null)
+            {
+                InitGenericInstances(genericParams, instanceArgs, tok);
```

Comment:
You should check that tok isn't null, or you can check in InitGenericInstances that your index is not out of bounds.